File: hw/sanchez/u3/hw17OPP/LibrarySystemPython/controller/book_controller.py
class BookController:

    def __init__(self, dao):
        self.dao = dao

    def save_book(self, book):
        self.dao.save(book)

    def get_all_books(self):
        return self.dao.find_all()

    def get_book_by_id(self, book_id):
        return self.dao.find_by_id(book_id)

    def update_book(self, book):
        self.dao.update(book)

    def delete_book(self, book_id):
        self.dao.delete(book_id)

File: hw/sanchez/u3/hw17OPP/LibrarySystemJavaScript/controller/BookController.js
class BookController {

    constructor(dao) {
        this.dao = dao;
    }

    async saveBook(book) {
        await this.dao.save(book);
    }

    async getAllBooks() {
        return await this.dao.findAll();
    }

    async getBookById(id) {
        return await this.dao.findById(id);
    }

    async updateBook(book) {
        await this.dao.update(book);
    }

    async deleteBook(id) {
        await this.dao.delete(id);
    }
}

module.exports = BookController;